graphqlapi/filters.go

```diff
@@ -4,179 +4,102 @@ import (
-}
 
-// generate the AND/OR/EQ/NEQ/IE filter fields for the ConvertedFetch and MetaFetch fields
-func genFilterFields(filterOptions map[string]*graphql.InputObject, filterFetchOptions map[string]*graphql.InputObject) graphql.InputObjectConfigFieldMap {
-
-	filterDescriptor := initializeFilterDescriptor()
+	for key, value := range staticFilterElements {
+		filterFields[key] = value
+	}
 
-	genSingleFilterFetchField(filterFetchOptions, "AND", filterDescriptor)
-	genSingleFilterFetchField(filterFetchOptions, "OR", filterDescriptor)
+	return filterFields
+}
 
-	inputType := graphql.InputObjectConfigFieldMap{
-		"AND": &graphql.InputObjectFieldConfig{
-			Type:        genSingleFilterField(filterOptions, filterFetchOptions, "AND", filterDescriptor),
-			Description: filterDescriptor.filterDescriptions["AND"],
+// generate these elements once
+func genStaticWhereFilterElements() graphql.InputObjectConfigFieldMap {
+	staticFilterElements := graphql.InputObjectConfigFieldMap{
+		"operator": &graphql.InputObjectFieldConfig{
+			Type:        genOperatorObject(),
+			Description: "Operator in the 'where' filter field, value is one of the 'WhereOperatorEnum' object",
 		},
-		"OR": &graphql.InputObjectFieldConfig{
-			Type:        genSingleFilterField(filterOptions, filterFetchOptions, "OR", filterDescriptor),
-			Description: filterDescriptor.filterDescriptions["OR"],
+		"path": &graphql.InputObjectFieldConfig{
+			Type:        graphql.NewList(graphql.String),
+			Description: "Path of from 'Things' or 'Actions' to the property name through the classes",
 		},
-		"EQ": &graphql.InputObjectFieldConfig{
-			Type:        graphql.NewList(genSingleFilterField(filterOptions, filterFetchOptions, "EQ", filterDescriptor)),
-			Description: filterDescriptor.filterDescriptions["EQ"],
+		"valueInt": &graphql.InputObjectFieldConfig{
+			Type:        graphql.Int,
+			Description: "Integer value where the property of the path will be compared to by an operator",
 		},
-		"NEQ": &graphql.InputObjectFieldConfig{
-			Type:        graphql.NewList(genSingleFilterField(filterOptions, filterFetchOptions, "NEQ", filterDescriptor)),
-			Description: filterDescriptor.filterDescriptions["NEQ"],
+		"valueFloat": &graphql.InputObjectFieldConfig{
+			Type:        graphql.Float,
+			Description: "Float value where the property of the path will be compared to by an operator",
 		},
-		"IE": &graphql.InputObjectFieldConfig{
-			Type:        graphql.NewList(genSingleFilterField(filterOptions, filterFetchOptions, "IE", filterDescriptor)),
-			Description: filterDescriptor.filterDescriptions["IE"],
+		"valueBoolean": &graphql.InputObjectFieldConfig{
+			Type:        graphql.Boolean,
+			Description: "Boolean value where the property of the path will be compared to by an operator",
+		},
+		"valueString": &graphql.InputObjectFieldConfig{
+			Type:        graphql.String,
+			Description: "String value where the property of the path will be compared to by an operator",
 		},
```

Comment:
This is not 100% in line with the data types that are supported, see the [data type definitions](https://github.com/creativesoftwarefdn/weaviate/blob/develop/schema/schema_handler.go#L61-L69)

In particular:
-  `valueFloat` does not match with `number`
- dates are missing

Reply:
Agreed, but this is how it is currently implemented in the Prototype. I can modify this, but that means the acceptance test(s) will fail. We'd need to update the Prototype first, that is the template for this code.

Dates were probably omitted because graphql-go has no Date object currently, it gets treated as a string. However, Alice the random user doesn't know this, so I agree that a Date option would help with ease of use.

Reply:
Then we need input from @bobvanluijt @laura-ham over here.

Reply:
Let's align this;

1. @laura-ham, can you please change this in the prototype.
2. @JeroenStravers1, can you align this in the tests and code based on @moretea's point and the updated prototype?

Reply:
I updated the `filters.go` file to reflect the data type definitions and updated the `schema_design.json` file to reflect these changes (ahead of the prototype). I have omitted the `"Unknown"` type as there is no such thing in GraphQL.

Reply:
After discussing this with @laura-ham I see that the filter options shouldn't have been renamed. The Date field should be omitted for now, as it currently does not fit in the design. I created an [issue](https://github.com/creativesoftwarefdn/weaviate/issues/482) to fix this.